sdk/dnsresolver/Azure.ResourceManager.DnsResolver/tests/DnsResolverTestBase.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.DnsResolver.Tests;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.Resources.Models;
+using Azure.ResourceManager.TestFramework;
+using Newtonsoft.Json;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.DnsResolver.Tests
+{
+    public class DnsResolverTestBase : ManagementRecordedTestBase<DnsResolverManagementTestEnvironment>
+    {
+        protected AzureLocation DefaultLocation => AzureLocation.WestUS2;
+        protected ArmClient Client { get; private set; }
+        protected Subscription DefaultSubscription { get; private set; }
+        public DnsResolverTestBase(bool isAsync) : base(isAsync)
+        {
+        }
+
+        public DnsResolverTestBase(bool isAsync, RecordedTestMode mode) : base(isAsync, mode)
+        {
+        }
+
+        [SetUp]
+        public async Task CreateCommonClient()
+        {
+            Client = GetArmClient();
+            DefaultSubscription = await Client.GetDefaultSubscriptionAsync().ConfigureAwait(false);
+        }
+
+        protected async Task<ResourceGroup> CreateResourceGroupAsync()
+        {
+            var resourceGroupName = Recording.GenerateAssetName("testRG-");
+            var rgOp = await DefaultSubscription.GetResourceGroups().CreateOrUpdateAsync(
+                true,
+                resourceGroupName,
+                new ResourceGroupData(DefaultLocation)
+                {
+                    Tags =
+                    {
+                        { "test", "env" }
+                    }
+                });
+            return rgOp.Value;
+        }
+
+        protected async Task CreateVirtualNetworkAsync(string virtualNetworkName)
```

Comment:
Can we add a project reference to `Azure.ResourceManager.Network` and use it to create the vnet?

Reply:
This is intentional. For recording, we have selected the ARM dogfood environment as we don't want to create so many resources in the prod environment. As of today, there is a limitation that virtual networks and subnets can't be created in the ARM dogfood. As a workaround, my team has written the simulator service to create VNETs and subnets. This function is calling that simulator service.